pxr/usd/lib/usd/stage.cpp

```diff
@@ -1732,6 +1732,36 @@ UsdStage::GetPrimAtPath(const SdfPath &path) const
     return UsdPrim(primData, proxyPrimPath);
 }
 
+UsdObject
+UsdStage::GetObjectAtPath(const SdfPath &path) const
+{
+    // Maintain consistent behavior with GetPrimAtPath
+    if (!path.IsAbsolutePath()) {
+        return UsdObject();
+    }
+
+    bool isPrimPath = path.IsPrimPath();
+    bool isPropPath = path.IsPropertyPath();
```

Comment:
Can we make these bools const? Also, can we avoid calling IsPropertyPath() if isPrimPath is already true:
```
const bool isPrimPath = path.IsPrimPath();
const bool isPropPath = !isPrimPath && path.IsPropertyPath();
```